packages/jest-mock/src/index.ts

```diff
@@ -1115,6 +1119,15 @@ export class ModuleMocker {
     return fn;
   }
 
+  private _attachMockImplementation<T extends Function>(
+    mock: Mock<UnknownFunction>,
+    original: T,
```

Comment:
`extends Function` is used here, because of `@typescript-eslint/ban-types`. It disallows using `Function` type directly, but is fine to see `extends`. In a way that makes sense, or?